src/gcp/cloudfunctions.ts

```diff
@@ -228,33 +219,74 @@ export async function createFunction(
   }
 }
 
-/**
- * @param name Fully qualified name of the Function.
- * @param policy The [policy](https://cloud.google.com/functions/docs/reference/rest/v1/projects.locations.functions/setIamPolicy) to set.
- */
-interface IamOptions {
-  name: string;
-  policy: any; // TODO: Type this?
+// getIamPolicy response body
+interface GetIamPolicy {
+  bindings?: iam.Binding[];
+  version?: number;
+  etag?: string;
 }
 
 /**
- * Sets the IAM policy of a Google Cloud Function.
- * @param options The Iam options to set.
+ * Gets the current invoker IAM policy for the function and overrides the invoker with the supplied invoker members
+ * @param projectId id of the project
+ * @param fnName function name
+ * @param invoker an array of invoker strings
+ *
+ * @throws {@link FirebaseError} on an empty invoker, when the IAM Polciy fails to be grabbed or set
  */
-export async function setIamPolicy(options: IamOptions): Promise<void> {
-  const endpoint = `/${API_VERSION}/${options.name}:setIamPolicy`;
+export async function setInvoker(
```

Comment:
Right now, I perform both api calls internally, would it be a better practice to separate the get & set calls into their own functions?